File: tst/blend.c
#include "pico.h"
#include "tst.c"

int main (void) {
    pico_init(1);
    pico_set_title("Blend");

    {
        puts("image dimming");
        Pico_RectX r = { 0.5, 0.5, 0.5, 0, PICO_ANCHOR_C, NULL };
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_output_draw_imageX(&r, "open.png");
            pico_input_delay(50);
            if (a == 120) {
                //_pico_check("image_dimmed");
            }
        }
    }
    {
        puts("pixel dimming");
        Pico_PosX p = { 0.5, 0.5, PICO_ANCHOR_NW, NULL };
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_set_color_draw((Pico_Color){255,0,0});
            pico_output_draw_pixelX(&p);
            pico_input_delay(50);
            if (a == 120) {
                _pico_check("pixel_dimmed");
            }
        }
    }
#if 0
    {
        puts("text dimming");
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_set_color_draw((Pico_Color){255,0,0});
            pico_output_draw_text(pos, "SOME TEXT");
            pico_input_delay(50);
            if (a == 120) {
                _pico_check("text_dimmed");
            }
        }
    }
#endif
    {
        puts("rect dimming");
        Pico_RectX r = { 0.5, 0.5, 0.9, 0.33, PICO_ANCHOR_C, NULL };
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_set_color_draw((Pico_Color){255,0,0});
            pico_output_draw_rectX(&r);
            pico_input_delay(50);
            if (a == 120) {
                //_pico_check("rect_dimmed");
            }
        }
    }
    {
        puts("oval dimming");
        Pico_RectX r = { 0.5, 0.5, 0.9, 0.33, PICO_ANCHOR_C, NULL };
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_set_color_draw((Pico_Color){255,0,0});
            pico_output_draw_ovalX(&r);
            pico_input_delay(50);
            if (a == 120) {
                //_pico_check("oval_dimmed");
            }
        }
    }
    {
        puts("tri dimming");
        Pico_PosX p1 = { 0.50, 0.33, PICO_ANCHOR_C, NULL };
        Pico_PosX p2 = { 0.33, 0.66, PICO_ANCHOR_C, NULL };
        Pico_PosX p3 = { 0.66, 0.66, PICO_ANCHOR_C, NULL };
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_set_color_draw((Pico_Color){255,0,0});
            pico_output_draw_triX(&p1, &p2, &p3);
            pico_input_delay(50);
            if (a == 120) {
                //_pico_check("tri_dimmed");
            }
        }
    }
    {
        puts("line dimming");
        Pico_PosX p1 = { 0.33, 0.33, PICO_ANCHOR_C, NULL };
        Pico_PosX p2 = { 0.66, 0.66, PICO_ANCHOR_C, NULL };
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_set_color_draw((Pico_Color){255,0,0});
            pico_output_draw_lineX(&p1, &p2);
            pico_input_delay(50);
            if (a == 120) {
                //_pico_check("line_dimmed");
            }
        }
    }
    {
        puts("poly dimming");
        Pico_PosX poly[] = {
            { 0.50, 0.33, PICO_ANCHOR_C, NULL },
            { 0.33, 0.66, PICO_ANCHOR_C, NULL },
            { 0.66, 0.66, PICO_ANCHOR_C, NULL },
        };
        for (int a = 255; a > 0; a-=5) {
            pico_output_clear();
            pico_set_alpha(a);
            pico_set_color_draw((Pico_Color){255,0,0});
            pico_output_draw_polyX(poly, 3);
            pico_input_delay(50);
            if (a == 120) {
                //_pico_check("poly_dimmed");
            }
        }
    }

    pico_init(0);
    return 0;
}

File: lua/tst/blend.lua
local pico = require 'pico'

pico.init(true)

pico.set.title "Blending..."

local pos = pico.pos(50,50)
local rct = {x=pos.x, y=pos.y, w=60, h=12}

do
    print "pixel dimming"
    for a=255, 0, -5 do
        pico.output.clear()
        pico.set.color.draw { r=255, g=0, b=0, a=a }
        pico.output.draw.pixel(pos)
        pico.input.delay(50)
        --[[
        if (a == 120) {
            _pico_check("pixel_dimmed")
        }
        ]]
    end
end

do
    print "text dimming"
    for a=255, 0, -5 do
        pico.output.clear()
        pico.set.color.draw { r=255, g=0, b=0, a=a }
        pico.output.draw.text(pos, "SOME TEXT");
        pico.input.delay(50)
        --[[
        if (a == 120) {
            _pico_check("text_dimmed")
        }
        ]]
    end
end

do
    print "rect dimming"
    for a=255, 0, -5 do
        pico.output.clear()
        pico.set.color.draw { r=255, g=0, b=0, a=a }
        pico.output.draw.rect(rct);
        pico.input.delay(50)
        --[[
        if (a == 120) {
            _pico_check("rect_dimmed")
        }
        ]]
    end
end

do
    print "oval dimming"
    for a=255, 0, -5 do
        pico.output.clear()
        pico.set.color.draw { r=255, g=0, b=0, a=a }
        pico.output.draw.oval(rct);
        pico.input.delay(50)
        --[[
        if (a == 120) {
            _pico_check("oval_dimmed")
        }
        ]]
    end
end

do
    print "tri dimming"
    for a=255, 0, -5 do
        pico.output.clear()
        pico.set.color.draw { r=255, g=0, b=0, a=a }
        pico.output.draw.tri(rct);
        pico.input.delay(50)
        --[[
        if (a == 120) {
            _pico_check("tri_dimmed")
        }
        ]]
    end
end

do
    print "line dimming"
    local p1 = { x=pos.x-20, y=pos.y-6 }
    local p2 = { x=pos.x+20, y=pos.y+6 }
    for a=255, 0, -5 do
        pico.output.clear()
        pico.set.color.draw { r=255, g=0, b=0, a=a }
        pico.output.draw.line(p1, p2);
        pico.input.delay(50)
        --[[
        if (a == 120) {
            _pico_check("line_dimmed")
        }
        ]]
    end
end

do
    print "poly dimming"
    local poly = {{x=5, y=5}, {x=59, y=10}, {x=20, y=31}};
    for a=255, 0, -5 do
        pico.output.clear()
        pico.set.color.draw { r=255, g=0, b=0, a=a }
        pico.output.draw.poly(poly)
        pico.input.delay(50)
        --[[
        if (a == 120) {
            _pico_check("poly_dimmed")
        }
        ]]
    end
end

pico.init(false)
